### PR TITLE
Create Periodic Filter

### PR DESCRIPTION
**Pull request by Twinters007**
This is a better solution to my previously proposed periodic filter. 

To use this, all that is needed is to add this filter to a logger with `self.logger.addFilter(PeriodicFilter(3))` and you're set. 

This filter isn't global, so each class will need it's own filter, but that allows for more customization, including bypass level

### CODE REVIEW

File: robotpy_ext/misc/periodic_filter.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import logging
+import wpilib
+
+
+class PeriodicFilter:
+    """
+        Periodic Filter to help keep down clutter in the console.
+        Simply add this filter to your logger and the logger will
+        only print periodically.
```
Comment (virtuald) [inline robotpy_ext/misc/periodic_filter.py]: Add example usage?

File: robotpy_ext/misc/periodic_filter.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+        self.period = period
+        self.loggingLoop = True
+        self._last_log = -period
+        self.bypassLevel = bypassLevel
+
+    def filter(self, record):
+        """Performs filtering action for logger"""
+        self._refresh_logger()
+        return self.parent.loggingLoop or record.levelno >= self.bypassLevel
```
Comment (virtuald) [inline robotpy_ext/misc/periodic_filter.py]: What sets `self.parent`?

**Comment by Twinters007** [comment]
Looking at this I clearly didn't test 👎

File: robotpy_ext/misc/periodic_filter.py (hunk 2 above)
Comment (Twinters007) [inline robotpy_ext/misc/periodic_filter.py]: 🤷‍♂️

**Pull request merged**